Formula/v2raya.rb

```diff
@@ -0,0 +1,35 @@
+class V2raya < Formula
+  desc "A web GUI client of Project V, which supports V2Ray, Xray, SS, SSR, Trojan and Pingtunnel."
+  homepage "https://v2raya.org"
+  url "https://github.com/v2rayA/v2rayA/archive/refs/tags/v1.5.7.zip"
+  sha256 "1a635783559fd4ef29fa99b2300d9f76c1d9c7638b12851288199a12e1676af5"
+  license "AGPL-3.0-Only"
+  head "https://github.com/v2rayA/v2rayA.git", branch: "master"
+
+  depends_on "go" => :build
+  depends_on "node@16" => :build
+  depends_on "yarn" => :build
+  depends_on "v2ray"
+
+  livecheck do
+    url :stable
+    strategy :github_latest
+  end
+
+  def install
+    ENV.deparallelize
+    system "bash", "-c", "export OUTPUT_DIR=../service/server/router/web; cd ./gui; yarn; yarn update; yarn build --outDir ../service/server/router/web/"
+    system "bash", "-c", "cd ./service/; go build -ldflags \"-X github.com/v2rayA/v2rayA/conf.Version=$version -s -w\" -o ../v2raya"
+    bin.install "v2raya"
+  end
+
+  test do
+    system "#{bin}/v2raya", "--version"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.